homeassistant/components/modbus/climate.py

```diff
@@ -268,21 +281,12 @@ async def async_set_hvac_mode(self, hvac_mode: HVACMode) -> None:
         """Set new target hvac mode."""
         if self._hvac_onoff_register is not None:
             # Turn HVAC Off by writing 0 to the On/Off register, or 1 otherwise.
-            if self._hvac_onoff_write_registers:
-                await self._hub.async_pb_call(
-                    self._slave,
-                    self._hvac_onoff_register,
-                    [0 if hvac_mode == HVACMode.OFF else 1],
-                    CALL_TYPE_WRITE_REGISTERS,
-                )
-            else:
-                await self._hub.async_pb_call(
-                    self._slave,
-                    self._hvac_onoff_register,
-                    0 if hvac_mode == HVACMode.OFF else 1,
-                    CALL_TYPE_WRITE_REGISTER,
-                )
-
+            await self._hub.async_pb_call(
```

Comment:
I'm just de-duplicating the code here by using a variable for the call type
Since the new setting is optional, without defaults, no installations will break